.clang-tidy

```diff
@@ -16,6 +16,16 @@ misc-*,\
 modernize-*,\
 performance-*,\
 readability-*,\
+-readability-named-parameter,\
```

Comment:
Might be worth considering disabling these until bazil = fast builds comes online in a week or two.

```
-readability-named-parameter -> "all parameters should be named in a function"
-readability-named-parameter -> "same as above"
-misc-unused-parameters -> "all parameters must be used - somehow this is triggered even with -WError"
-performance-unnecessary-value-param -> "parameter is copied for each invocation but only used as a const reference; consider making it a const reference"
-hicpp-use-auto -> "use auto when initializing with a cast to avoid duplicating the type name"
-modernize-use-auto -> "same as above"
-performance-move-const-arg -> "passing result of std::move() as a const reference argument; no move will actually happen"
-readability-container-size-empty -> "the 'empty' method should be used to check for emptiness instead of 'size'"
-readability-implicit-bool-conversion -> "implicit conversions to bool: if(pointer)"
-readability-inconsistent-declaration-parameter-name -> "Overloading functions"
```